### ISSUE TITLE
Accessibility Testing for Veteran Facing Forms - Form Upload

### ISSUE THREAD

**Issue by tbaker1026** [department-of-veterans-affairs/va.gov-team#90832]
# IN PROGRESS

### Product information

- [X] Team name, product name, and feature name have been added to the title of this issue.
- [X] Team label, product label and feature label (if applicable) have been added to this issue.

[Starting URL](https://staging.va.gov/form-upload/21-0779/introduction)

### Who completed the use of color and color contrast test?

@jeana-adhoc 

### Use of color and color contrast checks

- [X] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [X] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [X] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [X] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [X] Any charts, maps, infographics, and tables convey all information without only relying on color
- [X] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### How did color testing go?

No issues

### Who completed the axe scans?

@jeana-adhoc 

### axe checks

- [X] Each page has been scanned using axe (results shared below)
- [X] axe is integrated into your end-to-end testing

### axe DevTools scan results

| Axe Results | Page |
| -- | -- |
| No issues | [Introduction page](https://staging.va.gov/form-upload/21-0779/introduction) |
| No issues  | [Name & Zip Code](https://staging.va.gov/form-upload/21-0779/name-and-zip-code) |
| No issues | [Identification Info](https://staging.va.gov/form-upload/21-0779/identification-information) |
| No issues | [Phone & Email](https://staging.va.gov/form-upload/21-0779/phone-number-and-email) |
| No issues | [Upload](https://staging.va.gov/form-upload/21-0779/upload) |
| No issues | [Review](https://staging.va.gov/form-upload/21-0779/review-and-submit) |
| No issues | [Confirmation page](https://staging.va.gov/form-upload/21-0779/confirmation) |

### How did axe testing go?

Issues were initially identified, but have been fixed - those issues are attached to this ticket.

Ticket: https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/1949

### Who completed the content zoom and reflow test?

@jeana-adhoc 

### Content zoom and reflow checks

- [X] All page elements are readable and usable at 200% zoom
- [X] All page elements are readable and usable at 300% zoom
- [X] All page elements are readable and usable at 400% zoom

### How did content zoom and reflow testing go?

At 400% long file names break out of the bounding box of the form upload card. This is because the VA-card has a max-width defined on it at 75% and we needed `word-break:break-all` to handle underscores. This issue is attached and has been addressed

Ticket: https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/1950

### Who completed the keyboard navigation test?

@jeana-adhoc 

### Keyboard navigation checks

- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [X] All elements under focus have a visible focus indicator
- [X] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

No issues

### Do you have any other results to share?

1. As soon as I clicked upload, the error triggered. Selecting to upload the file should not trigger an error. I think the error should only trigger when a person tries to proceed (click continue) on the page and hasn't uploaded a file. There is a ticket for this already with the DST: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3625

<img width="504" alt="Image" src="https://github.com/user-attachments/assets/e7639832-9e62-4f8b-a791-a751293c28c2" />

2. On the upload screen, there seems to be a process in which the file is loading and a user cannot click on continue, then the page flashes and a user can continue.  This issue is documented here and with the Design system team: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3634

3. On the review page, edit links need accessible name
<img width="843" alt="Image" src="https://github.com/user-attachments/assets/1e62c82f-1d8e-494c-880c-45f5375172a8" />

Use label prop ([storybook](https://design.va.gov/storybook/?path=/docs/components-va-link--docs)), edit links should read "Edit personal information", "Edit contact information", etc.  Ticket: https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/1951

<br /><br />

4. I noticed that the "Zip code" label does not match what we have in the mailing address pattern, and should read "Postal code" and there is no validation on the field. I've filed an issue for this. Ticket: https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/1952

**Comment by jeana-adhoc** [comment]
This ticket is still in progress and will be completed Jan 2/3. I'm waiting on PRs to get merged in that will allow this ticket to mention far fewer defects.

**Comment by jeana-adhoc** [comment]
At the end of Sprint 23, I still can't get the confirmation page to load, so I can't test it. Will leave open for next sprint waiting for bugs to be addressed.

**Comment by jeana-adhoc** [comment]
Confirmation page is loading, and was tested, and no issues were discovered. This can be closed now.
